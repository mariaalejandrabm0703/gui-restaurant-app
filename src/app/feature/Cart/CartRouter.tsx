import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from '../../shared/components/LazyFallback';

const CartPage = React.lazy(() => import('./pages/Cart'));

export const CartRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    <Switch>
      <Route path="/" component={CartPage}></Route>
    </Switch>
  </React.Suspense>
);
