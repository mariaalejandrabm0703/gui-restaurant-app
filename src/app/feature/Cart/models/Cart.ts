export interface IClient {
    /**
     * Identificador del cliente
     */
    id?: number;
    /**
     * Nombre del cliente
     */
    nombre: string;
    /**
     * Identificacion del cliente
     */
    identificacion: string;
    /**
     * Telefono del cliente
     */
    telefono: string;
    /**
     * Estado del producto
     */
    activo?: string;
    /**
     * Email del cliente
     */
    email: string;
  }