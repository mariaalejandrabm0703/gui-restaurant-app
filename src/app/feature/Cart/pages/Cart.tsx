import * as React from 'react';
import { Layout } from 'app/shared/components/Layout';
import { ManageCart } from '../hoc/ManageCart';
import { RouteComponentProps } from 'react-router-dom';

const CartPage: React.FC<RouteComponentProps> = () => (
  <Layout title="Carrito" description="Carrito del cliente.">
    <ManageCart>Hola9</ManageCart>
  </Layout>
);

CartPage.displayName = 'CartPage';

export default CartPage;
