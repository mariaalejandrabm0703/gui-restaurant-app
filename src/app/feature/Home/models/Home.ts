export interface IProduct {
  /**
   * Identificador del producto
   */
  id: number;
  /**
   * Descripción del producto
   */
  descripcion: string;
  /**
   * Categoría del producto
   */
  categoria: string;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Imagen del producto
   */
  img: string;
}

export interface IProductOrderRegistred {
  /**
   * Identificador del producto
   */
  id: number;
 /**
   * Cantidad del producto
   */
  cantidad: number;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Producto del pedidoProducto
   */
  producto: IProduct;
}

export interface IProductOrder {
  /**
   * Identificador del producto
   */
   producto: number;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Cantidad del producto
   */
  cantidad: number;
}
export interface IRanking {
  /**
   * Identificador del producto
   */
  id: number;
  /**
   * Descripción del producto
   */
  descripcion: string;
  /**
   * Categoría del producto
   */
  categoria: string;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Imagen del producto
   */
  img: string;
  /**
   * Cantidad vendida del producto
   */
  cantidad: number;
}
