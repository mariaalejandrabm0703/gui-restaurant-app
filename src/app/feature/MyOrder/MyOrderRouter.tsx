import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from '../../shared/components/LazyFallback';

const MyOrderPage = React.lazy(() => import('./pages/MyOrderPage'));

export const MyOrderRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    <Switch>
      <Route path="/myOrder" component={MyOrderPage}></Route>
    </Switch>
  </React.Suspense>
);
