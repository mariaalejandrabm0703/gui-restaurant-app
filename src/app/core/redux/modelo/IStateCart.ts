import { IProductOrder } from '../../../feature/Home/models/Home';

export interface IStateCart {
  listProductsCart: IProductOrder[];
  count: number;
}
