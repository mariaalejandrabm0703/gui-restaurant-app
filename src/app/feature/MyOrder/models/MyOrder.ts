import { IProductOrder } from 'app/feature/Home/models/Home';

export interface IMyClient {
  /**
   * Identificador del cliente
   */
  id: number;
  /**
   * Nombre del cliente
   */
  nombre: string;
  /**
   * Indetificación del cliente
   */
  identificacion: string;
  /**
   * Telefono del cliente
   */
  telefono: string;
  /**
   * Email del cliente
   */
  email: string;
  /**
   * Estado del cliente
   */
  activo: string;
}
export interface IMyOrder {
  /**
   * Identificador del pedido
   */
  id: number | 0 ;
  /**
   * Descripción del pedido
   */
  fechaEntrega: string;
  /**
   * Precio del pedido
   */
  precio: number;
  /**
   * Estado del pedido
   */
  activo: string;
  /**
   * Productos
   */
  productosPedidos: IProductOrder[];
  /**
   * cliente del pedido
   */
  cliente: IMyClient;
}

export interface IMyOrderReg {
  /**
   * Descripción del pedido
   */
  fechaEntrega: string;
  /**
   * Precio del pedido
   */
  precio: number;
  /**
   * Estado del pedido
   */
  activo: string;
  /**
   * Productos
   */
  productos: IProductOrder[];
  /**
   * cliente del pedido
   */
  cliente: number;
}