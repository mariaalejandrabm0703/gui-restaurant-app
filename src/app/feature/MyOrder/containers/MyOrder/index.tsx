import * as PropTypes from 'prop-types';
import * as React from 'react';
import { IErrorToast } from '../../../../core/redux/modelo/IStateMain';
import { IMyOrder } from '../../models/MyOrder';
import ToastError from '../../../../shared/components/ToastError';
import {setOrderAsync} from '../../../../core/redux/actions/order/ActionsOrder';

interface OrderProps {
  myOrder: IMyOrder;
  isLoading: boolean;
  errorMessage: IErrorToast;
}

export const MyOrder: React.FC<OrderProps> = ({
  myOrder,
  isLoading,
   errorMessage,

}) => {
  // React.useEffect(() => {
  //   if (myOrder && errorMessage.message === '') {
  //     setOrderAsync(1);
  //   }
  // }, [myOrder]);

  return (
    <div className="container">
      <ToastError />
      <h1>Mi pedido</h1>
    </div>
  );
};
