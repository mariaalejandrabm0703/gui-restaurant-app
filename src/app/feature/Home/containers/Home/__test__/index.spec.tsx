import '@testing-library/jest-dom';
import {
  IFilters,
  IProduct,
  IProductOrder,
  IProductOrderRegistred,
  IRanking,
} from '../../../../Home/models/Home';
import { IMyClient, IMyOrder } from '../../../../MyOrder/models/MyOrder';
import {
  clientInfo,
  filtersInfo,
  myClientInfo,
  myOrderInfo,
  productInfo,
  productOrderInfor,
  productOrderRegistredInfo,
  rankingInfo,
} from '../../../../../shared/utils/data';
import { BrowserRouter } from 'react-router-dom';
import {Home as HomeC} from '../index';
import { IClient } from '../../../../Cart/models/Cart';
import { IErrorToast } from '../../../../../core/redux/modelo/IStateMain';
import { IStateCart } from '../../../../../core/redux/modelo/IStateCart';
import { IStateHome } from '../../../../../core/redux/modelo/IStateHome';
import { IStateMain } from '../../../../../core/redux/modelo/IStateMain';
import { IStateOrder } from '../../../../../core/redux/modelo/IStateOrder';
import { IStateProducts } from '../../../../../core/redux/modelo/IStateProducts';
import { IStateRanking } from '../../../../../core/redux/modelo/IStateRanking';
import { Provider } from 'react-redux';
import React from 'react';
import ReactDOM from 'react-dom';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const mockStore = configureStore([]);

const newProduct: IProduct = productInfo;
const listProducts: IProduct[] = [newProduct];
const products: IStateProducts = {
  listProducts,
};

const newRamking: IRanking = rankingInfo;
const listRanking: IRanking[] = [newRamking];
const ranking: IStateRanking = {
  listRanking,
};

const count: number = 1;
const client: IClient = clientInfo;
const modo: string = 'registry';
const newProductOrder: IProductOrder = productOrderInfor;
const listProductsCart: IProductOrder[] = [newProductOrder];
const cart: IStateCart = {
  listProductsCart,
  count,
  client,
  modo,
};

const main: IStateMain = {
  isLoading: false,
  errorMessage: { message: '', type: '' },
};

const newProductRegistred: IProductOrderRegistred = productOrderRegistredInfo;
const pedidosProductos: IProductOrderRegistred[] = [newProductRegistred];
const cliente: IMyClient = myClientInfo;
const myOrder: IMyOrder = {
  id: 1,
  fechaEntrega: '01/10/2021',
  precio: 500,
  activo: '1',
  pedidosProductos,
  cliente: cliente,
};
const order: IStateOrder = {
  myOrder,
};

const listProductsHome: IProduct[] = [newProduct];
const filters: IFilters = filtersInfo;
const home: IStateHome = {
  listProductsHome,
  filters,
};

const myOrderC: IMyOrder = myOrderInfo;
const isLoading: boolean = true;
const errorMessage: IErrorToast = { message: '', type: '' };
const searchOrderAsync = (id: number) => '';
const setProductsAsync = () => '';
const listProductsC: Array<IProduct> = [];

describe('Pruebas de container Home', () => {
  let store;

  let wrapper;
  beforeEach(() => {
    store = mockStore({
      products,
      ranking,
      cart,
      main,
      order,
      home,
    });

    wrapper = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <HomeC
            listProducts={[]}
            listRanking={[]}
            listProductsFilter={[]}
            isLoading={true}
            errorMessage={ { message: '', type: '' }}
            getAllProducts={()=> null}
            getProductsRanking={()=> null}
            setProductFilters={()=> null}
          />
        </BrowserRouter>
        ,
      </Provider>
    );
  });

  it('Compara snapshot del HomeC renderizado', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renderiza el componente HomeC y verifica que exista componente titulo', () => {
    const { getByTestId } = render(
      <Provider store={store}>
      <BrowserRouter>
        <HomeC
          listProducts={[]}
          listRanking={[]}
          listProductsFilter={[]}
          isLoading={true}
          errorMessage={ { message: '', type: '' }}
          getAllProducts={()=> null}
          getProductsRanking={()=> null}
          setProductFilters={()=> null}
        />
      </BrowserRouter>
      ,
    </Provider>
    );
    expect(getByTestId('div-home')).toContainElement(getByTestId('title-home'));
  });

  it('renderiza el componente HomeC y verifica que exista componente de ranking', () => {
    const { getByTestId } = render(
      <Provider store={store}>
      <BrowserRouter>
        <HomeC
          listProducts={[productInfo]}
          listRanking={[]}
          listProductsFilter={[]}
          isLoading={true}
          errorMessage={ { message: '', type: '' }}
          getAllProducts={()=> null}
          getProductsRanking={()=> null}
          setProductFilters={()=> null}
        />
      </BrowserRouter>
      ,
    </Provider>
    );
    expect(getByTestId('div-home')).toContainElement(getByTestId('div-ranking'));
  });
});
