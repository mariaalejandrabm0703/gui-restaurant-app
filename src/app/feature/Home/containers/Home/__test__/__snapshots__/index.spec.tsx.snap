// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de container Home Compara snapshot del HomeC renderizado 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <Home
      errorMessage={
        Object {
          "message": "",
          "type": "",
        }
      }
      getAllProducts={[Function]}
      getProductsRanking={[Function]}
      isLoading={true}
      listProducts={Array []}
      listProductsFilter={Array []}
      listRanking={Array []}
      setProductFilters={[Function]}
    />
  </BrowserRouter>
  ,
</ContextProvider>
`;
